src/app/ApplicationTemplate.Tests.Functional/FunctionalTestBase.cs

```diff
@@ -114,6 +118,31 @@ public TViewModel GetAndAssertActiveViewModel<TViewModel>()
 		return Assert.IsType<TViewModel>(ActiveViewModel);
 	}
 
+	/// <summary>
+	/// Simulates the back button press.
+	/// </summary>
+	public void NavigateBackUsingHardwareButton()
+	{
+		_backButtonSource.RaiseBackRequested();
+	}
+
+	class BackButtonSource : IBackButtonSource
```

Comment:
Make the class explicitly private and sealed.

Reply:
I completely forgot about that class, Do you think it's better to make that class private than to put it elsewhere? I guess it won't be used anywhere else

Reply:
Well if it's used and working as a private class, you can leave it. Are you getting any warning or suggestion from you IDE on this?